### PR TITLE
[WIP] WebExtension Support

### PR DESCRIPTION
**Pull request by kevincharm**
- [x] Figure out how to handle `browser-polyfill.js` - just import it
- [x] ~Should we resolve assets from code like `browser.extension.getURL('/some/asset.js')`?~
- [x] Should we auto-populate [`web_accessible_resources`](https://developer.mozilla.org/en-US/Add-ons/WebExtensions/manifest.json/web_accessible_resources)? - currently resolving as child bundles.
- [x] Tests

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -97,6 +97,8 @@
     "vue-template-compiler": "^2.5.16"
   },
   "scripts": {
+    "localinstall":
+      "npm uninstall -g parcel-bundler && yarn build && npm install -g .",
```
Comment (kevincharm) [inline package.json]: Remove

File: src/FSCache.js [hunk 2]
```diff
@@ -67,6 +67,11 @@ class FSCache {
   }
 
   async read(filename) {
+    const isWebExtensionManifest = filename.match(/manifest\.json$/);
```
Comment (kevincharm) [inline src/FSCache.js]: Add optargs check (`--webext` or something)

File: src/Parser.js [hunk 3]
```diff
@@ -6,6 +6,7 @@ const glob = require('glob');
 class Parser {
   constructor(options = {}) {
     this.extensions = {};
+    this.filenames = {};
```
Comment (kevincharm) [inline src/Parser.js]: better naming?

File: src/assets/WebExtensionManifestAsset.js [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    this.isAstDirty = false;
+  }
+
+  parse(code) {
+    return JSON.parse(code);
+  }
+
+  processSingleDependency(path, opts) {
+    opts = opts || {entry: true};
+    let assetPath = this.addURLDependency(path, opts);
```
Comment (kevincharm) [inline src/assets/WebExtensionManifestAsset.js:16]: const

File: src/Parser.js [hunk 5]
```diff
@@ -61,17 +66,30 @@ class Parser {
...
       return GlobAsset;
     }
 
+    // TODO: Clean this up.
+    const basename = path.basename(filename);
+    let parser = this.filenames[basename];
+    if (typeof parser === 'string') {
+      parser = this.filenames[filename] = require(parser);
+    }
+    if (parser) {
```
Comment (kevincharm) [inline src/Parser.js]: should maybe just check optargs for `--webext` and explicitly return the `WebExtensionManifestAsset` parser

**Comment by helpermethod** [comment]
browser-polyfill can be imported as an ES6 module, no? I don't think special handling is needed.

**Comment by kevincharm** [comment]
@helpermethod how are you using browser-polyfill atm? Right now I am loading it as a script via `manifest.json` or on the page itself.

**Comment by helpermethod** [comment]
I've added it as a dependency in `package.json` and importing it in my background/page_action/browser_action scrips like so

```js
import browser from 'webextension-polyfill'

browser.tabs.onUpdated.addListener((tabId, { status }, { url }) => {
  if (status === 'complete' && new URL(url).hostname.endsWith('myHostName')) {
    // TODO replace with browser.pageAction.show
    chrome.pageAction.show(tabId)
  }
})
```

That why I don't have to add it to `manifest.json`.

**Comment by kevincharm** [comment]
Great, I'll cross that off the list then! Thanks!

**Pull request closed without merging**